### DISCUSSION TITLE
Spatie\MediaLibrary\MediaCollections\Exceptions\UnreachableUrl

### DISCUSSION THREAD

**Discussion by dprhmk** [spatie/laravel-medialibrary#2927] [Q&A]
I have a problem with factories seeding img from faker imageUrl

<img width="876" alt="Снимок экрана 2022-05-18 в 22 26 21" src="https://user-images.githubusercontent.com/68992437/169140434-f8de0d69-7ce1-4425-bfd3-81ac6c08055b.png">

i use Sail, php 8.1, Laravel 8

<img width="878" alt="Снимок экрана 2022-05-18 в 22 24 11" src="https://user-images.githubusercontent.com/68992437/169140142-977dc064-bfca-43f5-ae91-69fd5d9e1405.png">

**Comment by AkioSarkiz** [comment]
i have too php 8.1, laravel 9

**Reply by AkioSarkiz** [reply]
@dprhmk i found the problem, it is not related to this package

```
<warning>PHP Warning:  fopen(https://via.placeholder.com/640x480.png/001144?text=Miss+qui): Failed to open stream: HTTP request failed! HTTP/1.1 403 Forbidden
```

**Reply by AkioSarkiz** [reply]
see https://picsum.photos/

**Reply by AkioSarkiz** [reply]
my solution 

AppServiceProvider.php
```php
Factory::macro('getImageUrl', function (int $width, int $height): string {
    return sprintf(
        'https://picsum.photos/%d/%d',
        $width,
        $height
    );
});
```

UserFactory.php
```php
$imageUrl = $this->getImageUrl(105, 145)
```

**Reply by rosyanxone** [reply]
What class does the Factory use?
